oar/core/shipment.py

```diff
@@ -782,3 +784,189 @@ def drop_bugs(self) -> tuple[list[str], list[str]]:
+        """Get current container image health status from Pyxis freshness grades
+        
+        Args:
+            grades: List of freshness grade objects from Pyxis API
+            
+        Returns:
+            Current health status (A, B, C, etc.) or "Unknown" if not found
+            
+        Note:
+            Filters grades to find the one with start_date before now
+            Uses Pyxis API's freshness attribute to determine health
+        """
+        
+        if not grades:
+            return "Unknown"
+            
+        now = datetime.now(timezone.utc)
+        # Get all grades that started before now and sort by start_date (newest first)
+        valid_grades = sorted(
+            [g for g in grades if datetime.fromisoformat(g["start_date"]) <= now],
+            key=lambda g: datetime.fromisoformat(g["start_date"]),
+            reverse=True
+        )
+        
+        if not valid_grades:
+            return "Unknown"
+            
+        # Return the most recent grade (first in the sorted list)
+        return valid_grades[0].get("grade", "Unknown")
+
+    def check_component_image_health(self) -> tuple[int, int, list[dict]]:
+        """Check container image health status for all components in shipment MRs
+        
+        Returns:
+            tuple: 
+                [0] total images scanned count,
+                [1] unhealthy components count,
+                [2] list of unhealthy components with grade info
+        """
+        unhealthy_components = []
+        total_scanned = 0
+        
+        for mr in self._mrs:
+            try:
+                components = self._get_components_from_shipment(mr)
+                for component in components:
+                    try:
+                        pull_spec = component.get("containerImage")
+                        if not pull_spec:
+                            continue
+                            
+                        digest = self._get_image_digest(pull_spec)
+                        grades = self._query_pyxis_freshness(digest)
+                        grade = self._get_current_image_health_status(grades)
+                        
+                        total_scanned += 1
+                        if grade and grade != "Unknown" and grade > "B":
+                            unhealthy_components.append({
+                                "name": component.get("name", "Unknown"),
+                                "grade": grade,
+                                "pull_spec": pull_spec
+                            })
+                    except Exception as e:
+                        logger.warning(f"Failed to check freshness for component: {str(e)}")
+                        continue
+            except Exception as e:
+                logger.error(f"Failed to process MR {mr.merge_request_id}: {str(e)}")
+                continue
+                
+        return total_scanned, len(unhealthy_components), unhealthy_components
```

Comment:
Could we use a class instead of tuple? It would be readable.
Do we need to return len(unhealthy_components), we can call the len when it is needed?

Reply:
code is updated. can you review again, thx